### PR TITLE
Rename editor preference section heading

### PR DESCRIPTION
**Pull request by gtsiolis**
## Description

Re-opening https://github.com/gitpod-io/gitpod/pull/7428 because the build failed due to the long branch name, see https://github.com/gitpod-io/gitpod/pull/7428#issuecomment-1004486424.

> This will update the IDE (Editor) preference section heading from _Default IDE_ to _Editor_.

> This can probably be further improved in https://github.com/gitpod-io/gitpod/issues/6602.

> See [relevant discussion](https://gitpod.slack.com/archives/C01KGM9BH54/p1641237326485600) (internal).

## How to test

Go to [`/preferences`](https://gitpod.io/preferences).

## Release Notes

```release-note
Rename editor preference section heading
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please ask for approval from **gtsiolis** and additionally assign **jeanp413** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jeanp413` in a comment when ready.

Associated issue: *#6602*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**
- **[installer/pkg/components/server/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/ide/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
@jeanp413 could you take another pass? Build will most probably pass now. Cc @akosyakov @loujaybee in case this it'd be better to close this for now.

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7437#issuecomment-1004872600" title="Approved">JanKoehnlein</a>*
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **jeanp413** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @jeanp413` in a comment when ready.

Associated issue: *#6602*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]
- **[installer/pkg/components/server/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/ide/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by jeanp413 — APPROVED** [review]
/lgtm

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 71bd566a0d4856d023b5a74447dc3a7647b75263</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7437#issuecomment-1004872600" title="Approved">JanKoehnlein</a>*, *<a href="https://github.com/gitpod-io/gitpod/pull/7437#pullrequestreview-843746394" title="LGTM">jeanp413</a>*

Associated issue: *#6602*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [JanKoehnlein]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [JanKoehnlein]
- ~~[installer/pkg/components/server/ide/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/installer/pkg/components/server/ide/OWNERS)~~ [jeanp413]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by gtsiolis** [comment]
Thanks @JanKoehnlein @jeanp413! 🧡

**Pull request merged**
